java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -94,22 +96,38 @@ private FlowComputation fork(SymbolicValue symbolicValue) {
     return new FlowComputation(symbolicValue, addToFlow, terminateTraversal);
   }
 
-  private void run(@Nullable final ExplodedGraph.Node currentNode, @Nullable final Symbol trackSymbol) {
-    if (currentNode == null || visited.contains(currentNode)) {
-      return;
+  private static class NodeSymbol {
+    final ExplodedGraph.Node node;
+    final Symbol trackSymbol;
+
+    public NodeSymbol(@Nullable ExplodedGraph.Node node, @Nullable Symbol trackSymbol) {
+      this.node = node;
+      this.trackSymbol = trackSymbol;
     }
-    visited.add(currentNode);
-
-    Symbol newTrackSymbol = trackSymbol;
-    if (currentNode.programPoint.syntaxTree() != null) {
-      newTrackSymbol = addFlowFromLearnedSymbols(currentNode, trackSymbol);
-      Stream<Constraint> learnedConstraints = addFlowFromLearnedConstraints(currentNode);
-      if (learnedConstraints.anyMatch(terminateTraversal)) {
-        return;
+  }
+
+  private void run(@Nullable final ExplodedGraph.Node node, @Nullable final Symbol trackSymbol) {
+    Deque<NodeSymbol> workList = new ArrayDeque<>();
+    workList.add(new NodeSymbol(node, trackSymbol));
+    while (!workList.isEmpty()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS135)